core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -742,53 +698,146 @@ private[history] object FsHistoryProvider {
   private val APPL_END_EVENT_PREFIX = "{\"Event\":\"SparkListenerApplicationEnd\""
 
   private val LOG_START_EVENT_PREFIX = "{\"Event\":\"SparkListenerLogStart\""
+
+  /** Current version of the data written to the listing database. */
+  private val CURRENT_LISTING_VERSION = 1L
```

Comment:
I'll add a more verbose comment; but this is basically me punting proper versioning to the next Spark release after this code is added. This version number is a "nuclear option"; if we break the data that is serialized to disk, we increase this number, and the new SHS will delete all old data and re-generated it from event logs.

I'm punting because there is no versioning issue in the first version; there's no existing data that the SHS might try to read.

I plan to take a closer look at versioning after all the initial PRs go in, but leaving this here gives us a choice in case there's a more urgent need to break things.